doc: Update multi partition doc (#898)
numaproj/numaflow: docs/user-guide/reference/multi-partition.md

Multi-partitioned edges are only supported for pipelines with JetStream as ISB. Please ensure
that the JetStream is provisioned with more nodes to support higher throughput.

> Since partitions are owned by the vertex, to create a multi-partitioned edge

Reviewer: ```suggestion
Since partitions are owned by the to-vertex, to create a multi-partitioned edge
```

> we need to configure the vertex to have multiple partitions.

Reviewer: ```suggestion
we need to configure the to-vertex to have multiple partitions.
```

> The following code snippet provides an example of how to configure a vertex (in this case, the `cat` vertex) to have multiple partitions, which enables it to read at a high TPS.

Reviewer: ```suggestion
The following code snippet provides an example of how to configure a vertex (in this case, the `cat` vertex) to have multiple partitions, which enables it (`cat` vertex) to read at a higher throughput.
```
